arm/Microsoft.OperationalInsights/workspaces/.bicep/nested_rbac.bicep

```diff
@@ -1,12 +1,11 @@
-param roleAssignment object
+param roleAssignmentObj object
 param builtInRoleNames object
 param logAnalyticsWorkspaceName string
 
-resource nested_rbac 'Microsoft.Storage/storageAccounts/providers/roleAssignments@2020-04-01-preview' = [for principalId in roleAssignment.principalIds: {
-  name: '${logAnalyticsWorkspaceName}/Microsoft.Authorization/${guid(logAnalyticsWorkspaceName, principalId, roleAssignment.roleDefinitionIdOrName)}'
+resource roleAssigment 'Microsoft.Storage/storageAccounts/providers/roleAssignments@2020-04-01-preview' = [for principalId in roleAssignmentObj.principalIds: {
```

Comment:
Change resource provider

Reply:
Fixed. Please check and resolve if you agree.